Cargo.toml

```diff
@@ -6,9 +6,10 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-rrplug = { version = "2.0.0", features = ["presence"] }
+rrplug = { git = "https://github.com/R2NorthstarTools/rrplug" }
```

Comment:
What's the current state of publishing to crates.io?

I'm fine with merging while still pointing to git repo but I'd kinda wanna see rrplug for v3 being on crates.io soon after so that it's more accessible to anyone developing plugins ^^

Reply:
I can publish it at any moment since it's basically done, but I would rather not.
The main is issue is that I don't quite know how much traits/generics I need over macros. 

currently it's mix of both. I want to remove as much macros as possible. Due to how rrplug has to interact with the plugin system and the engine for some cases are hard it's hard to decide what would be better.